### PR TITLE
Clean up legacy feature checks

### PR DESCRIPTION
**Pull request by AlanGriffiths**
I noticed a bunch of feature checks that are not needed for any supported series and cleaned them up

### CODE REVIEW

File: src/platform/graphics/CMakeLists.txt [hunk 1]
```diff
@@ -38,10 +38,6 @@ add_library(mirplatformgraphicscommon OBJECT
 mir_generate_protocol_wrapper(mirplatformgraphicscommon "zwp_" protocol/linux-dmabuf-unstable-v1.xml)
 
 if (DRM_VERSION VERSION_GREATER 2.4.107)
-  target_compile_definitions(
```
Comment (RAOF) [inline src/platform/graphics/CMakeLists.txt:41]: Is this correct? We're only checking for `drm >= 2.4.84`, and `drmGetFormatModifierName`  appeared in 2.4.107 as far as I can tell.

Comment (RAOF) [reply src/platform/graphics/CMakeLists.txt:41]: Hm, and you're not actually removing any of the `#ifdef MIR_HAVE_DRM_GET_MODIFIER_NAME`s?

**Review by RAOF — CHANGES_REQUESTED** [review]
I think the `ModiferName` changes are incorrect; the rest looks fine.

File: src/platform/graphics/CMakeLists.txt (hunk 1 above)
Comment (AlanGriffiths) [reply src/platform/graphics/CMakeLists.txt]: Oops. Dropped that commit as it is messed up

File: cmake/Doxygen.cmake [hunk 2]
```diff
@@ -1,13 +1,13 @@
 # Check if doxygen is present and add 'make doc' target
-find_package(Doxygen)
+find_package(Doxygen 1.8.0)
 
 option(
   BUILD_DOXYGEN
   "Build Doxygen documentation as part of the default build"
-  OFF
+  OFF`
```
Comment (RAOF) [inline cmake/Doxygen.cmake:7]: This looks like a spurious '`'?

**Review by RAOF — APPROVED** [review]
Other than the typo, looks good to me.

bors delegate=AlanGriffiths

**Pull request merged**